### PR TITLE
[bugfix][FlowEbos] make compile when Scalar != double.

### PR DESCRIPTION
**Pull request by dr-robertk**
This PR fixes some compile issues when Scalar is chosen not to be double but something else.

### CODE REVIEW

**Comment by andlaus** [comment]
seems good to me.

@atgeirr: the button is yours

**Comment by andlaus** [comment]
jenkins build this opm-material=263 update_data please

**Comment by andlaus** [comment]
sorry for "hijacking" this PR to update the reference solutions for OPM/opm-material#263. (I think opening a "dummy PR" for this purpose is worse.)

**Comment by atgeirr** [comment]
Annoyingly enough I forgot about this PR for a long time, and it got a bit stale. I can contribute to getting it in shape again, but have a few questions, which I put in the code.

File: opm/autodiff/BlackoilModelEbos.hpp [hunk 1]
```diff
@@ -619,15 +619,15 @@ namespace Opm {
...
+                Scalar& p = priVars[Indices::pressureSwitchIdx];
                 const double& dp_rel_max = dpMaxRel();
                 const int sign_dp = dp > 0 ? 1: -1;
                 p -= sign_dp * std::min(std::abs(dp), std::abs(p)*dp_rel_max);
                 p = std::max(p, 0.0);
 
                 // Saturation updates.
-                const double dsw = active_[Water] ? dx[cell_idx][Indices::waterSaturationIdx] : 0.0;
-                const double dxvar = active_[Gas] ? dx[cell_idx][Indices::compositionSwitchIdx] : 0.0;
+                const double dsw = active_[Water] ? dx[cell_idx][Indices::waterSaturationIdx] : Scalar(0);
```
Comment (atgeirr) [inline opm/autodiff/BlackoilModelEbos.hpp]: Should not the dsw etc. variables also be Scalar? As written this seems to force Scalar to be implicitly convertible to double, which may not be possible? Or is that an assumption that is made also elsewhere?

File: opm/autodiff/BlackoilModelEbos.hpp [hunk 2]
```diff
@@ -1266,14 +1268,28 @@ namespace Opm {
...
-                    matLawManager->oilWaterHysteresisParams(
-                            pcSwMdc_ow[cellIdx],
-                            krnSwMdc_ow[cellIdx],
-                            cellIdx);
-                    matLawManager->gasOilHysteresisParams(
-                            pcSwMdc_go[cellIdx],
-                            krnSwMdc_go[cellIdx],
-                            cellIdx);
+                    {
+                        // convert to Scalar and back
```
Comment (atgeirr) [inline opm/autodiff/BlackoilModelEbos.hpp]: Why is the back-and-forth conversion needed?

**Pull request closed without merging**